src/gtest/dict_test.cpp

```diff
@@ -0,0 +1,48 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "generated_wrappers.hpp"
+
+using namespace ::testing;
+
+extern "C" {
+    #include "dict.h"
+    extern dictType keylistDictType;
+}
+
+class DictTest : public ::testing::Test {
+    protected:
+        MockValkey mock;
+        RealValkey real;
+
+    void SetUp() override {
+        memset(&server, 0, sizeof(valkeyServer));
+        server.hz = CONFIG_DEFAULT_HZ;
+    }
+
+    void TearDown() override {
+
+    }
+};
+
+TEST_F(DictTest, testDictApis) {
```

Comment:
Let's create several different example tests.
* A simple test that just does something with EXPECT_EQ and ASSERT_EQ, showing the difference.  Maybe 3 different types (EQ, LE, ..) of expects and an assert.  Show EXPECT_STREQ also
* Show a test that uses a matcher.
* Show a test that mocks something.

Aim for really simple, really short, tests.